plugins/org.yakindu.sct.model.sexec/src/org/yakindu/sct/model/sexec/transformation/ReactionBuilder.xtend

```diff
@@ -146,6 +146,35 @@ class ReactionBuilder {
 		return execExit.reactSequence
 	}
 	
+	/**
+	 * Checks if a transition explicitly handles the specified exit. A transition is handling an exit 
+	 * if it has neither trigger nor guard and if the specified exit point is listed as a transition property or if no 
+	 * exit point is specified for the case of the exit named 'default'.
+	 */
+	def explicitlyHandlesExit(Transition it, String exitName) {
+		val exits = exitPoints
+		return 
+			   hasNoTrigger 
+			&& (('default'.equals(exitName) && exits.nullOrEmpty) 
+				|| exits.contains(exitName))
+	}
+	
+	/**
+	 * Returns the transition that handles the default exit for a state if it exists or null if not.
+	 * The default transition is that one that explicitly handles 'default'.
+	 */
+	def defaultExitTransition(State it) {
+		outgoingTransitions.filter( t | t.hasNoTrigger && t.explicitlyHandlesExit('default')).head
+	}
+	
+	/**
+	 * Returns the names of all exit points specified by a transition.
+	 */
+	def exitPoints(Transition it) {
+		 properties.filter(typeof(ExitPointSpec)).map( eps | eps.exitpoint).toList
```

Comment:
I may be wrong, but should the line not look like this?
```suggestion
		 properties.filter[ExitPointSpec].map[ eps | eps.exitpoint].toList
```
Especially the square brackets for `map`.